### PR TITLE
Supporting UDP addresses in allow listener

### PR DESCRIPTION
**Pull request by oxtoacart**
Needed for properly integrating KCP

cc: @uaalto @myleshorton @fffw

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -131,7 +132,16 @@ func (l *allowinglistener) Accept() (net.Conn, error) {
...
-	ip := conn.RemoteAddr().(*net.TCPAddr).IP.String()
+	ip := ""
+	remoteAddr := conn.RemoteAddr()
+	switch addr := remoteAddr.(type) {
+	case *net.TCPAddr:
+		ip = addr.IP.String()
+	case *net.UDPAddr:
+		ip = addr.IP.String()
+	default:
+		log.Errorf("Remote addr %v is of unknown type %v, unable to determine IP", remoteAddr, reflect.TypeOf(remoteAddr))
```
Comment (joesis) [inline server/server.go]: Can we skip calling `allow` here? Passing an empty `ip` has no benefit and may lead to unexpected behavior.

Comment (oxtoacart) [inline server/server.go]: Fixed.

**Comment by joesis** [comment]
LGTM

**Comment by oxtoacart** [comment]
@joesis Thanks for the review!

**Pull request merged**